pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -664,23 +666,48 @@ func KubernetesVersionExistsInCloudProfile(cloudProfile *gardencorev1beta1.Cloud
 	return false, gardencorev1beta1.ExpirableVersion{}, nil
 }
 
-// DetermineLatestKubernetesPatchVersion finds the latest Kubernetes patch version in the <cloudProfile> compared
+// DetermineLatestKubernetesPatchVersion finds the latest Kubernetes patch version for its minor version in the <cloudProfile> compared
 // to the given <currentVersion>. In case it does not find a newer patch version, it returns false. Otherwise,
 // true and the found version will be returned.
 func DetermineLatestKubernetesPatchVersion(cloudProfile *gardencorev1beta1.CloudProfile, currentVersion string) (bool, string, error) {
-	ok, newerVersions, _, err := determineNextKubernetesVersions(cloudProfile, currentVersion, "~")
+	ok, newerVersions, _, err := determineNextKubernetesVersions(cloudProfile.Spec.Kubernetes.Versions, currentVersion, "~")
 	if err != nil || !ok {
 		return ok, "", err
 	}
 	sort.Strings(newerVersions)
 	return true, newerVersions[len(newerVersions)-1], nil
 }
 
+// GetKubernetesVersionForPatchUpdate finds the latest Kubernetes patch version for its minor version in the <cloudProfile> compared
+// to the given <currentVersion>. Preview versions do not qualify for the kubernetes auto update. In case it does not find a newer patch version, it returns false. Otherwise,
+// true and the found version will be returned.
+func GetKubernetesVersionForPatchUpdate(versions []gardencorev1beta1.ExpirableVersion, currentVersion string) (bool, string, error) {
+	ok, _, versions, err := determineNextKubernetesVersions(versions, currentVersion, "~")
+	if err != nil || !ok {
+		return ok, "", err
+	}
+
+	var filteredVersions []string
+	for _, v := range versions {
+		if v.Classification != nil && *v.Classification == gardencorev1beta1.ClassificationPreview {
```

Comment:
There are valid use cases when we want to update to deprecated versions. E.g a Shoot has not the latest & deprecated patch K8s version. The highest patch version with the same minor is also deprecated (but has no expiration date). This can be the case if the K8 version is not maintained any more.

We want the Shoot to be updated to the latest & also deprecated patch version of the same minor.

Also an update to an already expired version should be possible for the same reasons. As stated in the docs, the Shoot then would receive consecutive version updates in the following maintenance windows.